lightning/src/ln/channelmanager.rs

```diff
@@ -3095,7 +3128,10 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 			let mut channel_state_lock = self.channel_state.lock().unwrap();
 			let channel_state = &mut *channel_state_lock;
 
-			for (short_chan_id, mut pending_forwards) in channel_state.forward_htlcs.drain() {
+			let mut forward_htlcs = HashMap::new();
+			mem::swap(&mut forward_htlcs, &mut self.forward_htlcs.lock().unwrap().drain().collect());
```

Comment:
```suggestion
			let mut forward_htlcs = HashMap::new();
			mem::swap(&mut forward_htlcs, &mut self.forward_htlcs.lock().unwrap());
```
avoids an allocation

Reply:
Nice, thanks :)!